maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,223 @@
+
+			if( $noOfFoundTags === 0 ) {
+				return false;
+			}
+
+			$foundTags = $results[0];
+			$foundTagsMapIds = $results[1];
+
+			return true;
+		} else {
+			$this->output( sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL );
+		}
+	}
+
+	public function hasTagsToRemove( $foundTagsMapIds, $mapsUsingTheTileset, &$tagsToRemoveArrayKeyIds ) {
+		foreach( $foundTagsMapIds as $key => $mapId ) {
+			if ( in_array( $mapId, $mapsUsingTheTileset ) ) {
+				$tagsToRemoveArrayKeyIds[] = $key;
+			}
+		}
+
+		return !empty( $tagsToRemoveArrayKeyIds );
+	}
+
+	public function removeImapTagFromArticle( $articleId, $stringWithTagToRemove ) {
+		$user = User::newFromName( 'WikiaBot' );
+		$article = Article::newFromID( $articleId );
+
+		if ( $user->getId() === 0 ) {
+			$this->output( 'Failed using WikiaBot user' . PHP_EOL );
+		}
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$oldContent = $article->getContent();
+			$newContent = str_replace( $stringWithTagToRemove, "", $oldContent );
+
+			$this->output( sprintf( 'Trying to edit article #%d', $articleId ) . PHP_EOL );
+			if( !self::isTest() ) {
+				$result = $article->doEdit(
+					$newContent,
+					wfMessage(
+						'realmap-deprecated-info',
+						'[[community:User_blog:DaNASCAT/Technical_Update:_August_29,_2016]]'
+					)->text(),
+					EDIT_FORCE_BOT,
+					false,
+					$user
+				);
+			} else {
+				$result = true;
+			}
+
+
+			if( $result ) {
+				$this->output( sprintf( "Removed <imap/> tags from article #%d", $articleId ) . PHP_EOL );
+			} else {
+				$this->output( sprintf( "Failed to update article #%d", $articleId ) . PHP_EOL );
+			}
+		} else {
+			$this->output( sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL );
+		}
+	}
+
+	public function isValidTilesSetId( $tilesSetId ) {
+		if ( !self::isNaturalNumber( $tilesSetId ) ) {
+			return false;
+		}
+
+		$res = $this->maps->getTileSet( $tilesSetId );
+
+		if ( !isset( $res['success'] ) || $res['success'] !== true ) {
+			$this->error( 'API call failure when looking for a tiles set #' . $tilesSetId . '.' . PHP_EOL );
+		}
+
+		$this->output( "Tiles' set #" . $tilesSetId . " found." . PHP_EOL );
+		return $res['content']->id == $tilesSetId;
+	}
+
+	public function execute() {
+		$this->app = F::app();
+		$this->mapsClientConfig = $this->app->wg->IntMapConfig;
+		$this->maps = new WikiaMaps( $this->mapsClientConfig );
+
+		self::$test = $this->hasOption( 'test' );
+
+		$cityId = $this->app->wg->CityId;
+		$tilesSetId = $this->getOption( 'tiles-set-id' );
+
+		if ( self::isTest() ) {
+			$this->output( 'Mode: test run' . PHP_EOL );
+		} else {
+			$this->output( 'Mode: normal run' . PHP_EOL );
+		}
+
+		if ( !$this->isValidTilesSetId( $tilesSetId ) ) {
+			$this->error( 'Invalid tiles-set-id. Try again.' . PHP_EOL, 1 );
+		}
+
+		$mapsUsingTheTileset = self::getMapsIdsUsingTileset( $cityId, $tilesSetId );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)